### ISSUE TITLE
[COT] Fatal Error on adding Coupons to manual order 

### ISSUE THREAD

**Issue by ankitguptaindia** [woothemes/woocommerce#34470]
### Prerequisites

- [ ] I have carried out troubleshooting steps and I believe I have found a bug.
- [ ] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

When adding Coupons to manual orders, the site throws Fatal Error. 

https://user-images.githubusercontent.com/6384299/186706402-5a965fe4-cf48-4e81-99bc-401e39c0f912.mp4

### Expected behavior

Coupons should apply without any error. 

### Actual behavior

Site has below error in Log and coupon adding screen stuck in progress for long time. 

`
[25-Aug-2022 15:25:25 UTC] PHP Fatal error:  Uncaught Exception: Unimplemented in /Users/ank/Local Sites/cotwoolocal/app/public/wp-content/plugins/woocommerce/src/Internal/DataStores/Orders/OrdersTableDataStore.php:1400
Stack trace:
#0 /Users/ank/Local Sites/cotwoolocal/app/public/wp-content/plugins/woocommerce/includes/class-wc-data-store.php(224): Automattic\WooCommerce\Internal\DataStores\Orders\OrdersTableDataStore->release_held_coupons(Object(Automattic\WooCommerce\Admin\Overrides\Order), true)
#1 /Users/ank/Local Sites/cotwoolocal/app/public/wp-content/plugins/woocommerce/includes/wc-order-functions.php(932): WC_Data_Store->__call('release_held_co...', Array)
#2 /Users/ank/Local Sites/cotwoolocal/app/public/wp-content/plugins/woocommerce/includes/abstracts/abstract-wc-order.php(1186): wc_update_coupon_usage_counts(3924)
#3 /Users/ank/Local Sites/cotwoolocal/app/public/wp-content/plugins/woocommerce/src/Internal/Orders/CouponsController.php(77): WC_Abstract_Order->apply_coupon('20off')
#4 /Users/ank/Local Sites/cotwoolocal/app/public/wp-content/plugins/woocommerce/src/Internal/Orders/CouponsController.php(25): Automattic\WooCommerce\Internal\Orders\CouponsController->add_coupon_discount(Array)
#5 /Users/ank/Local Sites/cotwoolocal/app/public/wp-content/plugins/woocommerce/includes/class-wc-ajax.php(1166): Automattic\WooCommerce\Internal\Orders\CouponsController->add_coupon_discount_via_ajax()
#6 /Users/ank/Local Sites/cotwoolocal/app/public/wp-includes/class-wp-hook.php(307): WC_AJAX::add_coupon_discount('')
#7 /Users/ank/Local Sites/cotwoolocal/app/public/wp-includes/class-wp-hook.php(331): WP_Hook->apply_filters('', Array)
#8 /Users/ank/Local Sites/cotwoolocal/app/public/wp-includes/plugin.php(476): WP_Hook->do_action(Array)
#9 /Users/ank/Local Sites/cotwoolocal/app/public/wp-admin/admin-ajax.php(187): do_action('wp_ajax_woocomm...')
#10 {main}
  thrown in /Users/ank/Local Sites/cotwoolocal/app/public/wp-content/plugins/woocommerce/src/Internal/DataStores/Orders/OrdersTableDataStore.php on line 1400
`

### Steps to reproduce

1. Follow this guide and make site COT testable [Custom Order Table testing guide](https://developer.woocommerce.com/2022/05/16/call-for-early-testing-custom-order-table-migrations/)
2. Go to WP Dashboard > WooCommerce > Orders > New order
3. Add Coupon in Apply Coupon Section
4.  Check the error in Log file
5. Order screen will stuck in progress mode

### WordPress Environment

<Details>
`
### WordPress Environment ###

WordPress address (URL): https://cotwoolocal.local
Site address (URL): https://cotwoolocal.local
WC Version: 6.9.0
REST API Version: ✔ 6.9.0
WC Blocks Version: ✔ 8.0.0
Action Scheduler Version: ✔ 3.4.0
Log Directory Writable: ✔
WP Version: 6.0.1
WP Multisite: –
WP Memory Limit: 256 MB
WP Debug Mode: ✔
WP Cron: ✔
Language: en_US
External object cache: –

### Server Environment ###

Server Info: nginx/1.16.0
PHP Version: 8.0.0
PHP Post Max Size: 1,000 MB
PHP Time Limit: 1200
PHP Max Input Vars: 4000
cURL Version: 7.54.0
LibreSSL/2.6.5

SUHOSIN Installed: –
MySQL Version: 5.7.28
Max Upload Size: 300 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 6.9.0
WC Database Prefix: wp_
Total Database Size: 53.95MB
Database Data Size: 31.54MB
Database Index Size: 22.41MB
wp_woocommerce_sessions: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_api_keys: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_attribute_taxonomies: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_downloadable_product_permissions: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_order_items: Data: 0.13MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_order_itemmeta: Data: 1.52MB + Index: 0.72MB + Engine InnoDB
wp_woocommerce_tax_rates: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_tax_rate_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_shipping_zones: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_shipping_zone_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_shipping_zone_methods: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_payment_tokens: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_payment_tokenmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_log: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_actions: Data: 1.02MB + Index: 0.59MB + Engine InnoDB
wp_actionscheduler_claims: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_groups: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_logs: Data: 1.02MB + Index: 0.31MB + Engine InnoDB
wp_commentmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_comments: Data: 7.52MB + Index: 9.09MB + Engine InnoDB
wp_email_log: Data: 0.08MB + Index: 0.00MB + Engine InnoDB
wp_gla_budget_recommendations: Data: 0.22MB + Index: 0.14MB + Engine InnoDB
wp_gla_merchant_issues: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_gla_shipping_rates: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_gla_shipping_times: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_links: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_mo2f_network_blocked_ips: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_mo2f_network_email_sent_audit: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_mo2f_network_transactions: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_mo2f_network_whitelisted_ips: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_mo2f_user_details: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_mo2f_user_login_info: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_options: Data: 7.33MB + Index: 0.13MB + Engine InnoDB
wp_postmeta: Data: 7.42MB + Index: 8.75MB + Engine InnoDB
wp_posts: Data: 1.50MB + Index: 0.42MB + Engine InnoDB
wp_termmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_terms: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_term_relationships: Data: 0.11MB + Index: 0.06MB + Engine InnoDB
wp_term_taxonomy: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_usermeta: Data: 0.08MB + Index: 0.06MB + Engine InnoDB
wp_users: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wcpv_commissions: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wcpv_per_product_shipping_rules: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_admin_notes: Data: 0.06MB + Index: 0.00MB + Engine InnoDB
wp_wc_admin_note_actions: Data: 0.05MB + Index: 0.02MB + Engine InnoDB
wp_wc_bookings_availability: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_bookings_availabilitymeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_booking_relationships: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_category_lookup: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_customer_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_deposits_payment_plans: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_deposits_payment_plans_schedule: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_download_log: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_orders: Data: 0.28MB + Index: 0.13MB + Engine InnoDB
wp_wc_orders_meta: Data: 1.52MB + Index: 0.52MB + Engine InnoDB
wp_wc_order_addresses: Data: 0.17MB + Index: 0.14MB + Engine InnoDB
wp_wc_order_coupon_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_order_operational_data: Data: 0.14MB + Index: 0.06MB + Engine InnoDB
wp_wc_order_product_lookup: Data: 0.09MB + Index: 0.06MB + Engine InnoDB
wp_wc_order_stats: Data: 0.08MB + Index: 0.05MB + Engine InnoDB
wp_wc_order_tax_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_product_attributes_lookup: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_product_download_directories: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_product_meta_lookup: Data: 0.08MB + Index: 0.09MB + Engine InnoDB
wp_wc_rate_limits: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_reserved_stock: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_tax_rate_classes: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_webhooks: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_square_customers: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wpns_attack_logs: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wpns_backup_report: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wpns_files_scan: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wpns_ip_rate_details: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wpns_malware_hash_file: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wpns_malware_scan_report: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wpns_malware_scan_report_details: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wpns_malware_skip_files: Data: 0.02MB + Index: 0.00MB + Engine InnoDB

### Post Type Counts ###

attachment: 55
bookable_person: 34
bookable_resource: 150
event_ticket: 140
event_ticket_email: 1
global_product_addon: 1
nav_menu_item: 6
page: 40
post: 15
product: 368
product_variation: 34
revision: 240
shop_coupon: 3
shop_order: 626
shop_order_placehold: 4
shop_order_refund: 5
shop_subscription: 3
wc_booking: 620
wc_membership_plan: 1
wc_user_membership: 1
wp_global_styles: 2
wp_template_part: 1

### Security ###

Secure connection (HTTPS): ✔
Hide errors from visitors: ❌Error messages should not be shown to visitors.

### Active Plugins (2) ###

WooCommerce Product Vendors: by WooCommerce – 2.1.64
WooCommerce: by Automattic – 6.9.0-dev

### Inactive Plugins (28) ###

Bookings Helper: by WooCommerce – 1.0.3
Debug Log - Config Tool: by Lukman Nakib – 1.1
Email Log: by Sudar – 2.4.8
PHP Compatibility Checker: by WP Engine – 1.6.0
Pig Latin: by Nikolay Bachiyski – 0.2
Plugin Toggle: by Cedaro – 1.3.1
Publisher Media Kit: by 10up – 1.2.1
RTL Tester: by <a href="http://blog.yoavfarhi.com">Yoav Farhi</a>
<a href="http://automattic.com">Automattic</a> – 1.2

WooCommerce Accommodation Bookings: by WooCommerce – 1.1.33
WooCommerce Blocks: by Automattic – 8.3.1
WooCommerce Bookings: by WooCommerce – 1.15.60
WooCommerce Bookings Availability: by WooCommerce – 1.1.20
WooCommerce Box Office: by WooCommerce – 1.1.39
WooCommerce Currency Converter: by WooCommerce – 1.6.28
WooCommerce Deposits: by WooCommerce – 1.5.11
WooCommerce EU VAT Number: by WooCommerce – 2.4.2
WooCommerce Memberships: by SkyVerge – 1.22.9
WooCommerce Min/Max Quantities: by WooCommerce – 3.0.0
WooCommerce Multi-currency: by TIV.NET INC – 2.14.3
WooCommerce PayFast Gateway: by WooCommerce – 1.4.24
WooCommerce PDF Invoices: by Andrew Benbow – 4.15.6
WooCommerce Pre-Orders: by WooCommerce – 1.7.1
WooCommerce Product Add-ons: by WooCommerce – 4.4.0
WooCommerce Product CSV Import Suite: by WooCommerce – 1.10.48
WooCommerce Stripe Gateway: by WooCommerce – 6.6.0
WooCommerce Subscriptions: by WooCommerce – 3.1.3
WP Crontrol: by John Blackbourn & crontributors – 1.14.0
WP Rollback: by Impress.org – 1.7.1

### Settings ###

API Enabled: –
Force SSL: –
Currency: USD ($)
Currency Position: left
Thousand Separator: ,
Decimal Separator: .
Number of Decimals: 2
Taxonomies: Product Types: accommodation-booking (accommodation-booking)
booking (booking)
external (external)
grouped (grouped)
simple (simple)
subscription (subscription)
variable (variable)
variable subscription (variable-subscription)

Taxonomies: Product Visibility: exclude_from_catalog (exclude_from_catalog)
exclude_from_search (exclude_from_search)
exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)

Connected to WooCommerce.com: –
Enforce Approved Product Download Directories: –

### WC Pages ###

Shop base: #6 - /shop/
Cart: #2333 - /block-based-cart/
Checkout: #1940 - /block-based-checkout/
My account: #9 - /my-account/
Terms and conditions: ❌ Page not set

### Theme ###

Name: Storefront
Version: 4.1.2 (update to version 4.1.3 is available)
Author URL: https://woocommerce.com/
Child Theme: ❌ – If you are modifying WooCommerce on a parent theme that you did not build personally we recommend using a child theme. See: How to create a child theme
WooCommerce Support: ✔

### Templates ###

Overrides: –

### Admin ###

Enabled Features: activity-panels
analytics
coupons
customer-effort-score-tracks
experimental-products-task
experimental-import-products-task
experimental-fashion-sample-products
experimental-product-tour
shipping-smart-defaults
shipping-setting-tour
homescreen
marketing
mobile-app-banner
navigation
onboarding
onboarding-tasks
remote-inbox-notifications
remote-free-extensions
payment-gateway-suggestions
shipping-label-banner
subscriptions
store-alerts
transient-notices
wc-pay-promotion
wc-pay-welcome-page

Disabled Features: minified-js
new-product-management-experience
settings

Daily Cron: ✔ Next scheduled: 2022-08-26 13:39:06 +05:30
Options: ✔
Notes: 98
Onboarding: completed

### Action Scheduler ###

Complete: 1,577
Oldest: 2022-07-26 13:21:20 +0530
Newest: 2022-08-25 20:44:04 +0530

Pending: 3
Oldest: 2022-08-25 20:49:04 +0530
Newest: 2022-09-18 20:50:03 +0530

### Status report information ###

Generated at: 2022-08-25 20:49:28 +05:30
`
</Details>

### Isolating the problem

- [ ] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [ ] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).
- [ ] I can reproduce this bug consistently using the steps above.

**Comment by jeffpaul** [comment]
@alefesouza FYI ^
